src/base/net/geoipmanager.cpp

```diff
@@ -88,10 +85,8 @@ GeoIPManager *GeoIPManager::instance()
 
 void GeoIPManager::loadDatabase()
 {
-    if (m_geoIPDatabase) {
-        delete m_geoIPDatabase;
-        m_geoIPDatabase = nullptr;
-    }
+    delete m_geoIPDatabase;
+    m_geoIPDatabase = nullptr;
```

Comment:
@glassez 
maybe we should replace those raw pointers (only those we assign `nullptr` explicitly) with `std::unique_ptr`? So we don't have to remember to set it back to `nullptr` on deletion.

Reply:
IMO, it's not a job for `std::unique_ptr`.

Reply:
Since I was talking from memory safety viewpoint: https://isocpp.org/wiki/faq/freestore-mgmt#memory-leaks
I also saw C++ FAQ is advocating `make_unique` (and therefore `std::unique_ptr`) all over the place.. there are 7 matches on that page.

>IMO, it's not a job for std::unique_ptr.

I guess we won't see such changes in near future..

Reply:
Just to add to this and it is probably out-of-scope for this PR: Lately I have been watching many CPPCON videos on youtube. The general consensus seems to be: Don't have a `new` anywhere. Unless there is a good reason to. Don't even have `new` when constructing a smart pointer (aka use the `make_*` functions wherever possible).
Quite a few of our classes use QPointer already.
IMO, there should be a future migration to smart pointers in the whole codebase. (QPointer is ok too in classes deriving from Qt classes).

Reply:
>Quite a few of our classes use QPointer already.
IMO, there should be a future migration to smart pointers in the whole codebase. (QPointer is ok too in classes deriving from Qt classes).

Have I misunderstood QPointer? From the [docs](https://doc.qt.io/qt-5/qpointer.html#dtor.QPointer):
```
QPointer::~QPointer()
Destroys the guarded pointer. Just like a normal pointer, destroying a guarded pointer does not destroy the object being pointed to.
```

So it isn't equivalent to unique_ptr. So why do we use it?

Reply:
>IMO, there should be a future migration to smart pointers in the whole codebase. 

IMO most of the time we should prefer std pointer utilities over Qt ones.

>So it isn't equivalent to unique_ptr. So why do we use it?

See `m_aboutDlg` usage in mainwindow.cpp. It is occasionally useful that the object is already owned by some parent and when the object is deleted the QPointer will reset itself to nullptr.

From https://doc.qt.io/qt-5/qpointer.html#details
>Note that Qt 5 introduces a slight change in behavior when using QPointer.
>When using QPointer on a QWidget (or a subclass of QWidget), previously the QPointer would be cleared by the QWidget destructor. Now, the QPointer is cleared by the QObject destructor (since this is when QWeakPointer objects are cleared). Any QPointers tracking a widget will NOT be cleared before the QWidget destructor destroys the children for the widget being tracked.

Reply:
>IMO, there should be a future migration to smart pointers in the whole codebase. 

Do you even want to drop using of QObject memory management model?

Reply:
>Do you even want to drop using of QObject memory management model?

No. From what I read online it is discouraged to use std smart_pointers with QObject classes. They have different owning/lifetime models and they can't mix well.

Unless you meant something else with your question.